### PR TITLE
Generator - RandomTest

### PR DESCRIPTION
**Pull request by zhikezhang**
This PR creates the RandomTest. 
1. Add RandomTest, generating dapp automatically. This method has random number of arguments. The arguments of this method have randum types and random values. This dapp returns the abi-encoded arguments.
2. Also, Deploy, call and verify the randomly generated dapp.

### CODE REVIEW

File: test/IntegTest.java [hunk 1]
```diff
@@ -163,4 +167,5 @@ public void testFallback() {
         newVal = (Integer) callStatic(dapp, "getValue");
         assertEquals(oldVal + 20, newVal);
     }
+
```
Comment (arajasek) [inline test/IntegTest.java:161]: All of the changes to this file are redundant, so this file should not be included as part of the PR.

File: test/RandomTest.java [hunk 2]
```diff
@@ -0,0 +1,308 @@
+
+import java.io.ByteArrayInputStream;
+import java.io.DataOutputStream;
```
Comment (arajasek) [inline test/RandomTest.java]: Unused imports can be taken out.

File: test/RandomTest.java [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+    @Test
+    public void testArgumentsWithFixedTypesRandomValues() {
+        RandomArgumentsGenerator argsGenerator = new RandomArgumentsGenerator();
+        argsGenerator.addFixedTypesAndRandomValues(18);
+
+        String className = "RandomClass";
+
+        String methodName = argsGenerator.getRandomString(9);
+
+        System.out.println(String.format("%s.%s%s --- %s arguments", className , methodName,
```
Comment (arajasek) [inline test/RandomTest.java]: I would comment out this `println`, and the ones for the parameters. I think so long as you print out the seed, we should always be able to recreate the random test, which is all we want.

It gets irritating to have very verbose tests.

**Comment by zhike-aion** [comment]
Also, Re-organize the test folder, improve the code structures, etc..

File: test/RandomTest.java [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+                null,
+                null);
+        AnnotationVisitor annotationVisitor = methodVisitor.visitAnnotation("Lorg/aion/abigenerator/Callable;", false);
+        annotationVisitor.visitEnd();
+        methodVisitor.visitCode();
+
+        Label label0 = new Label();
+        methodVisitor.visitLabel(label0);
+        methodVisitor.visitLdcInsn("");
+
```
Comment (arajasek) [inline test/RandomTest.java:156]: Please insert comments explaining what the generated method(s?) do.

**Review by arajasek — CHANGES_REQUESTED** [review]
Thanks for working on this! I have some initial requested changes, might have more on a second review.

File: test/RandomTest.java (hunk 3 above)
Comment (zhikezhang) [reply test/RandomTest.java]: Good point!

File: test/RandomTest.java (hunk 2 above)
Comment (zhikezhang) [reply test/RandomTest.java]: Good point!

File: test/IntegTest.java (hunk 1 above)
Comment (zhikezhang) [reply test/IntegTest.java:161]: Good point!

**Review by arajasek — APPROVED** [review]
Thanks for the changes, looks good to me!

**Pull request merged**